### ISSUE TITLE
Feature request: Position scaling parameter

### ISSUE THREAD

**Issue by Frostion** [pushrax/OpenVR-SpaceCalibrator#23]
In addition to the X/Y/Z translation and rotation parameters already available in Space Calibrator, X/Y/Z position scaling/multiplation parameters would be very useful.

In my particular case, my WMR headset seems to under-detect position movement, so the farther I walk away from the location I calibrated at, the less accurate the positions of my Vive trackers get in relation to my head and hands. The only way to really fix this would be to have the ability to scale down the lighthouse playspace.

**Comment by pushrax** [comment]
Interesting, that can definitely be implemented. It can probably be auto-calibrated as well, I just assumed this wouldn’t be necessary.

**Comment by sundansx** [comment]
I think I also have this problem with my HTC Vive/stock controllers and my Samsung Odyssey+.  the calibration seems really good when at the original point the calibration stopped, but when I spin or move across my space, the controllers shift some.

**Comment by Nagorak** [comment]
Hey guys, I also had really bad "tracking drift" when using WMR with Steam VR tracking. However, I finally realized it was because my Pitch and Roll values were not being detected properly, resulting in the two tracking planes being misaligned. 

If you're experiencing bad tracking drift, check what your Pitch and Roll values are set to, and try rounding them to the nearest 90 degrees (i.e. if your Roll is -174.something, change it to -180, same for Pitch). That may fix your problem. (It could also help to change your Yaw in the same way, but try setting Pitch and Roll first, and then manually re-aligning your controllers.)

**Comment by Nagorak** [comment]
I will add that even with everything dialed in, location tracking does vary slightly between the two systems, so some sort of scaling parameter would still be quite useful.

**Comment by Amtrix** [comment]
What's the state of this? Considering getting index controllers for reverb g2 and this looks troubling.

**Comment by Nagorak** [comment]
In my experience, it works well enough to not be a problem, provided you have the pitch and roll values dialed in 100%. I've been running with Index Controllers/Vive Wands with my Reverb, and while there is technically some minor drift, it's not enough to matter in actual use.

**Comment by jgr526** [comment]
The problem seems to vary per WMR headset as shown in closed issue #45 (was merged with this one by Pushrax). Some people with WMR headsets have no noticeable scaling issue while others express seeing the same issue in their setups. My odyssey has little scaling problems, but my G2 has bad scaling issue.

https://youtu.be/l2jhvU0e-4Q

Issue #45 shows everything attempted to mitigate the issue, including Pitch/Roll tweaking via manual settings. Unfortunately I have not found anyway to remediate it via software, settings, or environment changes. The fact that my odyssey displays a much lower scaling error tells me its something hardware related. Hopefully the "old code" that Pushrax referenced in issue #45 might alleviate this issue for the people who have it/notice it.

**Comment by eclair4151** [comment]
+1 for me. Tried manually calibrating to get it perfect but as soon as I start walking around my play space it starts drifting. As someone has stated earlier it is super obvious if I go in the center put my hands out in front of me, and spin in a circle, controllers start oscillating towards and away from me.

**Comment by Kevin-Lewis** [comment]
I also have this same issue with controller drift when I move away from the center of my play space. What's weird is that I didn't have it for the first few days I set it up but since last night, it's constant.

After recalibrating, the drift will usually occur at a slightly different angle but to about the same magnitude.

**Comment by eclair4151** [comment]
@pushrax sorry to be a pain but any ideas on when we might expect a potential build with this  :D   I think this option would really help out, (in addition to allowing you to define the center of the playspace maybe so the scale knows to go from the center.

**Comment by pushrax** [comment]
Looks like there's lots of interest in this. Porting the scaling-aware solver shouldn't be too tricky though it's also not trivial. The high level tasks are:

- Update the driver to support a scale factor, or maybe just change it to accept a matrix instead of rotation/translation/scale. I'm thinking that extracting the scale and keeping it as a separate parameter will be slightly easier to reason about.
- Grab the [scale-aware solving code from the old simpler-solve branch](https://github.com/pushrax/OpenVR-SpaceCalibrator/commit/9040063e92f982684355f555f73730c053427281#diff-36df2c445dfc2699da70bc439075083abfbde2d01dc4e2a82d3c57f28540759cR301-R355) and integrate it in Calibration.cpp. I think the math can stay the same, except that the scale should be calculated from the 3x3 `rot` matrix. If we assuming the scale is isotropic, the scale factor is the length of any vector from the matrix (not sure if this is a good assumption or not...)

Defining the center of the playspace shouldn't be necessary; the translation between spaces already implicitly defines this.

**Comment by tacticalbacon89** [comment]
Yes please give us this update soon please i want to walk around my play space and not be locked to one spot not even able to spin around

**Comment by tacticalbacon89** [comment]
wait i think hes telling us to do it ourselves basically, and what needs to be done for it to work ummm yea i dont know enough about coding to accomplish this task without help and i dont think most people wanting to use this software do either

**Comment by TheCyberQuake** [comment]
I'm seeing this issue on Quest 2 as well. Trackers drift proportionally the further away from center I go. Seems like it's exactly what's described here

**Comment by pushrax** [comment]
There is now a scale setting available in v1.3 https://github.com/pushrax/OpenVR-SpaceCalibrator/releases

It's not automatically calibrated at the moment, but you can try experimenting with it.

**Comment by brian-armstrong** [comment]
I made some directions on how to use the new scale parameter on the post below

https://old.reddit.com/r/MixedVR/comments/qpj22e/new_official_space_calibrator_release_possible/
